src/sqlancer/yugabyte/ysql/oracle/YSQLNoRECOracle.java

```diff
@@ -0,0 +1,165 @@
+package sqlancer.yugabyte.ysql.oracle;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import sqlancer.IgnoreMeException;
+import sqlancer.Randomly;
+import sqlancer.common.oracle.NoRECBase;
+import sqlancer.common.oracle.TestOracle;
+import sqlancer.common.query.SQLQueryAdapter;
+import sqlancer.common.query.SQLancerResultSet;
+import sqlancer.yugabyte.ysql.YSQLCompoundDataType;
+import sqlancer.yugabyte.ysql.YSQLErrors;
+import sqlancer.yugabyte.ysql.YSQLGlobalState;
+import sqlancer.yugabyte.ysql.YSQLSchema;
+import sqlancer.yugabyte.ysql.YSQLSchema.YSQLColumn;
+import sqlancer.yugabyte.ysql.YSQLSchema.YSQLDataType;
+import sqlancer.yugabyte.ysql.YSQLSchema.YSQLTable;
+import sqlancer.yugabyte.ysql.YSQLSchema.YSQLTables;
+import sqlancer.yugabyte.ysql.YSQLVisitor;
+import sqlancer.yugabyte.ysql.ast.YSQLCastOperation;
+import sqlancer.yugabyte.ysql.ast.YSQLColumnValue;
+import sqlancer.yugabyte.ysql.ast.YSQLExpression;
+import sqlancer.yugabyte.ysql.ast.YSQLJoin;
+import sqlancer.yugabyte.ysql.ast.YSQLPostfixText;
+import sqlancer.yugabyte.ysql.ast.YSQLSelect;
+import sqlancer.yugabyte.ysql.gen.YSQLExpressionGenerator;
+import sqlancer.yugabyte.ysql.oracle.tlp.YSQLTLPBase;
+
+public class YSQLNoRECOracle extends NoRECBase<YSQLGlobalState> implements TestOracle {
+
+    private final YSQLSchema s;
+
+    public YSQLNoRECOracle(YSQLGlobalState globalState) {
+        super(globalState);
+        this.s = globalState.getSchema();
+        YSQLErrors.addCommonExpressionErrors(errors);
+        YSQLErrors.addCommonFetchErrors(errors);
+    }
+
+    public static List<YSQLJoin> getJoinStatements(YSQLGlobalState globalState, List<YSQLColumn> columns,
+            List<YSQLTable> tables) {
+        List<YSQLJoin> joinStatements = new ArrayList<>();
+        YSQLExpressionGenerator gen = new YSQLExpressionGenerator(globalState).setColumns(columns);
+        for (int i = 1; i < tables.size(); i++) {
```

Comment:
Just remembered https://github.com/sqlancer/sqlancer/issues/551. Perhaps this would be something to fix in a subsequent PR (perhaps after @ankushdesai (or someone else) fixed this for Postgres?